xgi/algorithms/connected.py

```diff
@@ -214,35 +261,63 @@ def largest_connected_hypergraph(H, in_place=False):
     connected_components
     largest_connected_component
 
+    References
+    ----------
+    Aksoy, S.G., Joslyn, C., Ortiz Marrero, C. et al.
+    Hypernetwork science via high-order hypergraph walks.
+    EPJ Data Sci. 9, 16 (2020).
+    https://doi.org/10.1140/epjds/s13688-020-00231-0
+
     Example
     -------
     >>> import xgi
     >>> H = xgi.random_hypergraph(10, [0.1, 0.01], seed=1)
     >>> H_gcc = xgi.largest_connected_hypergraph(H)
     >>> print(H_gcc.num_nodes)
     6
-
     """
-    connected_nodes = max(connected_components(H), key=len)
+    connected_nodes = max(connected_components(H, s=s), key=len)
     if not in_place:
         return subhypergraph(H, nodes=connected_nodes).copy()
     else:
         H.remove_nodes_from(set(H.nodes).difference(connected_nodes))
 
 
-def _plain_bfs(H, source):
-    """A fast BFS node generator
+def _plain_bfs(H, source, s=1):
+    """A helper function to do an edge-first BFS search
+
+    Parameters
+    ----------
+    H : xgi.Hypergraph
+        The hypergraph
+    source : hashable
+        The ID of the starting node for the BFS
+    s : int, optional
+        The overlap of edges, by default 1
 
-    Source:
+    Returns
+    -------
+    set
+        A list of all nodes in the s-component.
+
+    References
+    ----------
     https://networkx.org/documentation/stable/_modules/networkx/algorithms/components/connected.html
+
+    Aksoy, S.G., Joslyn, C., Ortiz Marrero, C. et al.
+    Hypernetwork science via high-order hypergraph walks.
+    EPJ Data Sci. 9, 16 (2020).
+    https://doi.org/10.1140/epjds/s13688-020-00231-0
     """
-    seen = set()
-    nextlevel = {source}
+    seen_edges = set()
+    nodes = {source}
+    nextlevel = set(H.edges(H.nodes.memberships(source)).filterby("size", s, "geq"))
```

Comment:
Maybe split into a few lines to make help readability?

Reply:
also maybe a brief comment describing what "nextlevel" and "thislevel" are? Set of neighbor nodes?
And "level" is the level of the BFsearch I guess?